src/Kernel.php

```diff
@@ -59,4 +72,133 @@ public function registerContainerConfiguration(LoaderInterface $loader): void
             }
         });
     }
+
+    protected function initializeContainer(): void {
+        parent::initializeContainer();
+        $this->container->set('pdg.configuration', $this->loadConfiguration());
+    }
+
+    public function loadConfiguration(): Configuration {
+        $cwd = getcwd();
+
+        // First, load config file from PDG_CONFIG_FILE environment variable
+        $configFile = getenv('PDG_CONFIG_FILE');
+        if ($configFile && !is_file($configFile)) {
+            throw new RuntimeException(sprintf('Configuration file "%s" does not exist.', $configFile));
+        }
+
+        // If PDG_CONFIG_FILE environment variable is not set, try to load config file from default ordered ones
+        if (!$configFile) {
+            $files = [
+                'pdg.config.yaml',
+                'pdg.config.yml',
+                'pdg.config.dist.yaml',
+                'pdg.config.dist.yml',
+            ];
+
+            foreach ($files as $filename) {
+                if (is_file(Path::join($cwd, $filename))) {
+                    $configFile = $filename;
+                    break;
+                }
+            }
+        }
+
+        // No config file detected
+        if (!$configFile) {
```

Comment:
If no file exists, we should still call `getConfigTree` to retrieve the default values and inject them in the Configuration object